### PR TITLE
Update CCD Web boot shell script: be able to login

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Fix ccd docker for local testing](https://tools.hmcts.net/jira/browse/BPS-620)

### Change description ###

1st attempt (3 commits with "Update" prefix): check configs and do some updates after looking into actual services and example setup
2nd hit - doh! S2S keys have been renamed
~3rd and last login fix which was head-walled multiple times in order to finally achieve: script update to create user profile in ccd environment~
definition upload creates all user profiles found in xslx so no need to create a profile manually. Creating personal caseworker for convenience

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by doncem** [comment]
In order to successfully import there is a need of following change in `hmcts/ccd-definition-store-api` as library returns `null` for `((XSSFCell) cell).getRawValue()`
```diff
diff --git a/excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/SpreadsheetParser.java b/excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/SpreadsheetParser.java
index 594bed47..b85a53db 100644
--- a/excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/SpreadsheetParser.java
+++ b/excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/SpreadsheetParser.java
@@ -7,7 +7,6 @@ import org.apache.poi.ss.usermodel.DataFormatter;
 import org.apache.poi.ss.usermodel.DateUtil;
 import org.apache.poi.ss.usermodel.Row;
 import org.apache.poi.ss.usermodel.Workbook;
-import org.apache.poi.xssf.usermodel.XSSFCell;
 import org.apache.poi.xssf.usermodel.XSSFWorkbook;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -62,7 +60,8 @@ public class SpreadsheetParser {
                     boolean validRow = false;
                     while (!validRow && cellIterator.hasNext()) {
                         Cell cell = cellIterator.next();
-                        if (cell != null && cell.getCellTypeEnum() != CellType.BLANK && !((XSSFCell) cell).getRawValue().isEmpty()) {
+
+                        if (cell != null && cell.getCellTypeEnum() != CellType.BLANK && !cell.toString().isEmpty()) {
                             validRow = true;
                         }
                     }
```

I'll propose PR

File: app-gateway.tf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 data "azurerm_key_vault" "infra_vault" {
-  name = "infra-vault-${var.subscription}"
-  resource_group_name = "${var.env == "prod" ? "core-infra-prod" : "cnp-core-infra"}"
+  name = "${var.env == "prod" ? "infra-cert-prod" : "infra-vault-${var.subscription}"}"
```
Comment (nitinprabhu) [inline app-gateway.tf]: I am sure these changes are not part of this PR ? could you please rebase ?

File: bin/start-ccd-web.sh [hunk 2]
```diff
@@ -44,11 +45,30 @@ az acr login --name hmcts --subscription ${SUBSCRIPTION_ID}
...
+
+while [[ `docker ps -a | grep starting | wc -l | awk '{$1=$1};1'` != "0" ]]
+do
+    echo "Waiting for "`docker ps -a | grep starting | wc -l | awk '{$1=$1};1'`" container(s) to start. Sleeping for 5 seconds..."
+    sleep 5
+done
+
+# for convenience, let's have personal caseworker account
+
+curl -XPOST \
```
Comment (nitinprabhu) [inline bin/start-ccd-web.sh]: Will this work if you run `star-ccd-web.sh` after is being executed once ? I assume it will throw an error that user already exists ?

Comment (nitinprabhu) [reply bin/start-ccd-web.sh]: I would say these could go into separate scripts and include instructions in readme to execute those whenever required

File: docker-compose.yml [hunk 3]
```diff
@@ -106,27 +106,30 @@ services:
...
+          service-auth-provider-api:
+            condition: service_started
         healthcheck:
           disable: true
         ports:
           - 4453:4453
 
     ccd-definition-store-api:
-      image: hmcts/ccd-definition-store-api:latest
+      image: hmcts.azurecr.io/hmcts/ccd-definition-store-api:pr-402
```
Comment (nitinprabhu) [inline docker-compose.yml]: comment would be best why we we are using tagged image and a todo :)

File: docker-compose.yml [hunk 4]
```diff
@@ -239,28 +261,21 @@ services:
         - IDAM_BASE_URL=http://idam-api:8080
         - IDAM_USER_URL=http://idam-api:8080
         - IDAM_LOGOUT_URL=https://localhost:8000/login/logout
+        - IDAM_OAUTH2_CLIENT_ID=ccd_gateway
+        - IDAM_OAUTH2_CLIENT_SECRET=123456
         - IDAM_OAUTH2_TOKEN_ENDPOINT=http://idam-api:8080/oauth2/token
         - IDAM_S2S_URL=http://service-auth-provider-api:8080
         - IDAM_SERVICE_KEY=AAAAAAAAAAAAAAAA
         - IDAM_SERVICE_NAME=ccd_gw
```
Comment (nitinprabhu) [inline docker-compose.yml]: this may be not required anymore ?

File: app-gateway.tf (hunk 1 above)
Comment (doncem) [reply app-gateway.tf]: hmm.. let me actually rebase it. i did the merge locally, not sure why this tf file got screwed. [never merge - just rebase :D]

File: bin/start-ccd-web.sh (hunk 2 above)
Comment (doncem) [reply bin/start-ccd-web.sh]:
```bash
$ ./bin/start-ccd-web.sh 
    "name": "Donatas.Martinkus@HMCTS.NET",
You are logged into HMCTS Azure Portal
Logging into the HMCTS Azure Container Registry
Login Succeeded
Starting docker containers...
bulk-scan-shared-infrastructure_smtp-server_1 is up-to-date
bulk-scan-shared-infrastructure_service-auth-provider-api_1 is up-to-date
bulk-scan-shared-infrastructure_ccd-api-gateway_1 is up-to-date
bulk-scan-shared-infrastructure_shared-database-v10_1 is up-to-date
bulk-scan-shared-infrastructure_idam-api_1 is up-to-date
bulk-scan-shared-infrastructure_ccd-user-profile-api_1 is up-to-date
Starting bulk-scan-shared-infrastructure_ccd-importer_1 ... 
Starting bulk-scan-shared-infrastructure_ccd-importer_1       ... done
Starting bulk-scan-shared-infrastructure_dm-store_1                ... done
Starting bulk-scan-shared-infrastructure_authentication-web_1 ... done
Starting bulk-scan-shared-infrastructure_idam-importer_1      ... done
Starting bulk-scan-shared-infrastructure_ccd-case-management-web_1 ... done
Starting bulk-scan-shared-infrastructure_ccd-data-store-api_1      ... done
Waiting for 1 container(s) to start. Sleeping for 5 seconds...
Waiting for 1 container(s) to start. Sleeping for 5 seconds...
... some waits here
Waiting for 1 container(s) to start. Sleeping for 5 seconds...
Waiting for 1 container(s) to start. Sleeping for 5 seconds...
User with this email account already exists
```
and script ends

Comment (nitinprabhu) [reply bin/start-ccd-web.sh]: ok that's fine still separating them gives ability to create it separately so bonus :)

File: docker-compose.yml (hunk 4 above)
Comment (doncem) [reply docker-compose.yml]: still [here](https://github.com/hmcts/ccd-api-gateway/blob/b11caa87a0d6bbd6484ba5c0b5c9868dad6a9918/config/custom-environment-variables.yaml#L12)

File: bin/start-ccd-web.sh (hunk 2 above)
Comment (doncem) [reply bin/start-ccd-web.sh]: bonuses are my thing, right? 🙂
